kolibri/core/assets/src/core-app/urls.js

```diff
@@ -4,12 +4,17 @@
  */
 
 import setWebpackPublicPath from '../utils/setWebpackPublicPath';
+import getPluginData from 'kolibri.utils.getPluginData';
+
+function generateUrl(baseUrl, url) {
+  const base = new URL(baseUrl, window.location.origin);
+  const urlObject = new URL(url, base);
+  return urlObject.href;
+}
 
 const urls = {
   setUp() {
-    Object.assign(this, global.kolibriUrls);
-    this.__staticURL = global.staticUrl;
-    this.__mediaURL = global.mediaUrl;
+    Object.assign(this, getPluginData().urls);
```

Comment:
This is a bit confusing because it's not clear what's actually being added to `this`.

Reply:
Mostly the `django-js-reverse` url definitions, will add comments.